plugins/batterystat.koplugin/main.lua

```diff
@@ -64,59 +63,64 @@ function Usage:percentagePerHour()
 end
 
 function Usage:remainingHours()
+    if self:percentagePerHour() == 0 then return "n/a" end
     local curr = State:new()
     return curr.percentage / self:percentagePerHour()
 end
 
 function Usage:chargingHours()
+    if self:percentagePerHour() == 0 then return "n/a" end
     local curr = State:new()
-    return (curr.percentage - 100) / self:percentagePerHour()
+    return math.abs(curr.percentage - 100) / self:percentagePerHour()
 end
 
 local function shorten(number)
+    if number == "n/a" then return _("n/a") end
     return string.format("%.2f", number);
 end
 
 function Usage:dump(kv_pairs)
-    table.insert(kv_pairs, {_("    Consumed %"), shorten(self.percentage)})
-    table.insert(kv_pairs, {_("    Total minutes"), shorten(self:minutes())})
-    table.insert(kv_pairs, {_("    % per hour"), shorten(self:percentagePerHour())})
+    table.insert(kv_pairs, {INDENTATION .. _("Consumed %"), shorten(self.percentage)})
+    table.insert(kv_pairs, {INDENTATION .. _("Total time"), util.secondsToHClock(self.time, true, true)})
+    table.insert(kv_pairs, {INDENTATION .. _("% per hour"), shorten(self:percentagePerHour())})
 end
 
 function Usage:dumpRemaining(kv_pairs)
-    table.insert(kv_pairs, {_("    Estimated remaining hours"), shorten(self:remainingHours())})
+    table.insert(kv_pairs, {INDENTATION .. _("Estimated remaining hours"), shorten(self:remainingHours())})
 end
 
 function Usage:dumpCharging(kv_pairs)
-    table.insert(kv_pairs, {_("    Estimated hours for charging"), shorten(self:chargingHours())})
+    table.insert(kv_pairs, {INDENTATION .. _("Estimated hours for charging"), shorten(self:chargingHours())})
 end
 
 local BatteryStat = {
     settings = LuaSettings:open(DataStorage:getSettingsDir() .. "/batterstat.lua"),
```

Comment:
Can we fix that filename to "battery_stats.lua" ?
(I guess we don't need to care about migration for such transient stuff.)

Reply:
Of course :)